core/config/project_settings.cpp

```diff
@@ -1524,6 +1556,43 @@ bool ProjectSettings::has_global_group(const StringName &p_name) const {
 	return global_groups.has(p_name);
 }
 
+void ProjectSettings::add_shader_global_parameter(const StringName &p_name, RS::GlobalShaderParameterType p_type, const Variant &p_value) {
+	if (RenderingServer::get_singleton()->global_shader_parameter_get(p_name).get_type() != Variant::NIL) {
+		ERR_FAIL_MSG(vformat("Global shader parameter '%s' already exists.", p_name));
+	}
+
+	List<String> keywords;
+	ShaderLanguage::get_keyword_list(&keywords);
+
+	if (keywords.find(p_name) != nullptr || p_name == "script") {
```

Comment:
```suggestion
	if (p_name == CoreStringName(script) || keywords.find(p_name) != nullptr) {
```
Though the check for `script` should be moved up to the very top